### PR TITLE
Move static build to GitHub actions

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
This moves the whole static build pipeline from CircleCI to GitHub
actions, except the integration tests. The integration tests will follow
at some later point, if other integration tests are available and the
overall test strategy is clear.
#### Which issue(s) this PR fixes:

Refers to #4489 
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
- Added `crio-status.8` man page to static release bundle
```

### CODE REVIEW

**Comment by haircommander** [comment]
I think test failures are unrelatred https://github.com/cri-o/cri-o/issues/4509

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -43,6 +55,7 @@ jobs:
       - run: |
           chmod -R +x bin
           sudo rm /etc/containers/storage.conf
+      - run: |
```
Comment (kolyshkin) [inline .github/workflows/test.yml:58]: What is the reason for this change?

**Review by kolyshkin — APPROVED** [review]
LGTM

**Comment by kolyshkin** [comment]
So I took a look at the artifacts produced (https://github.com/cri-o/cri-o/actions/runs/501561670), and it looks like `build` and `build-static` are identical, except `build-static` has `pinns` while `build` doesn't.

Also it probably make sense to set some retention policy, and/or remove some intermediate artifacts.

File: .github/workflows/test.yml (hunk 1 above)
Comment (saschagrunert) [reply .github/workflows/test.yml:58]: My intention to not name the steps was to be able to see what is going to be executed from the jobs overview page. Unfortunately it only displays the first line from multi-line executions. By adding another run step, It now looks like this: 
screenshot

Seems more clear to me what is going on.

**Comment by saschagrunert** [comment]
> So I took a look at the artifacts produced (https://github.com/cri-o/cri-o/actions/runs/501561670), and it looks like `build` and `build-static` are identical, except `build-static` has `pinns` while `build` doesn't.

`build` contains the dynamically linked binaries which are mainly going to be used for further validations and later on the integration tests.

`build-static` contains the statically linked binaries, which are intended to be directly consumed by users via the static binary bundle we provide for each commit/release.

> Also it probably make sense to set some retention policy, and/or remove some intermediate artifacts.

Right now we have an artifact and log retention set to 90 days via the project configuration settings:

screenshot
Do you have anything else in mind?
